### DISCUSSION TITLE
How to create GitHub issues for NPM deprecation notices

### DISCUSSION THREAD

**Discussion by p5** [renovatebot/renovate#28815] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I would like to have GitHub issues created when a package I am using is marked as "deprecated" in NPM.  

After reading the docs, we believe all we should need to do is set `suppressNotifications` to an empty array, since it defaults to `["deprecationWarningIssues"]` (see [docs](https://docs.renovatebot.com/configuration-options/#suppressnotifications)).  
After changing this setting, we are able to see Renovate identifying the deprecation notice in the logs, however no issues are being created, nor are any errors being shown in the job logs.

This package is directly mentioned in my `package.json` as a `dependency`, but is also a sub-dependency of other packages I am using.

Here is Renovate's outputs for the package that I expect to have created the issue.

<details>

```json
{
            "currentValue": "3.374.0",
            "currentVersion": "3.374.0",
            "datasource": "npm",
            "depName": "@aws-sdk/signature-v4",
            "depType": "dependencies",
            "deprecationMessage": "On registry `https://**redacted**@aws-sdk/signature-v4` has the following deprecation notice:\n\n`This package has moved to @smithy/signature-v4`\n\nMarking the latest version of an npm package as deprecated results in the entire package being considered deprecated, so contact the package author you think this is a mistake.",
            "fixedVersion": "3.374.0",
            "homepage": "https://github.com/aws/aws-sdk-js-v3/tree/main/packages/signature-v4",
            "lockedVersion": "3.374.0",
            "packageName": "@aws-sdk/signature-v4",
            "prettyDepType": "dependency",
            "registryUrl": "https://registry.npmjs.org/",
            "sourceDirectory": "packages/signature-v4",
            "sourceUrl": "https://github.com/aws/aws-sdk-js-v3",
            "versioning": "npm",
            "warnings": [],
            "updates": []
          },
...
DEBUG: Found deprecationMessage for npm package @aws-sdk/signature-v4
```

</details>

And please see my current configuration below:

```jsonc
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "rebaseWhen": "never",
  // Prefer Dependabot Security Alerts
  "vulnerabilityAlerts": {
    "enabled": false
  },  
  "suppressNotifications": []
}
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

          {
            "currentValue": "3.374.0",
            "currentVersion": "3.374.0",
            "datasource": "npm",
            "depName": "@aws-sdk/signature-v4",
            "depType": "dependencies",
            "deprecationMessage": "On registry `https://**redacted**@aws-sdk/signature-v4` has the following deprecation notice:\n\n`This package has moved to @smithy/signature-v4`\n\nMarking the latest version of an npm package as deprecated results in the entire package being considered deprecated, so contact the package author you think this is a mistake.",
            "fixedVersion": "3.374.0",
            "homepage": "https://github.com/aws/aws-sdk-js-v3/tree/main/packages/signature-v4",
            "lockedVersion": "3.374.0",
            "packageName": "@aws-sdk/signature-v4",
            "prettyDepType": "dependency",
            "registryUrl": "https://registry.npmjs.org",
            "sourceDirectory": "packages/signature-v4",
            "sourceUrl": "https://github.com/aws/aws-sdk-js-v3",
            "versioning": "npm",
            "warnings": [],
            "updates": []
          },
```

```
DEBUG: Found deprecationMessage for npm package @aws-sdk/signature-v4
```

</details>

**Comment by rarkins** [comment]
We essentially disabled this feature as long ago there was occasional "spam" problems with too many issues raised, people not understanding it, etc. I think that one issue per deprecated dependency is probably too noisy. How about listing them in the Dependency Dashboard issue?

**Reply by p5** [reply]
Thank you for the reply!

> How about listing them in the Dependency Dashboard issue?

That would definitely be helpful! There's currently no way in either GitHub, GitHub Advanced Security or Renovate of seeing these deprecation notices without bringing in additional dependencies, so being able to see notices in one place would be great.
